### PR TITLE
Change check() function to test actual types in Series and Index

### PR DESCRIPTION
**Pull request by Dr-Irv**
While reviewing #372 , it uncovered that by having `check()` use `next(iter())`, if you have a `Series([1,2,3], dtype=np.int8)`, it was not checking that the actual type of the first element was `np.int8`, since `Series.__iter__()` returns a python scalar.

So this corrects that, allowing us to be more precise in the type checking of the values inside the results, since we want to check that `astype()` is doing what we expect.

Also, was able to allow `pytables` to work with 3.11, since they released a compatible version.

Uncovered an odd issue with bitwise operations on lists of `int`

In some cases, used `np.integer` because it is a supertype that works across platforms.  Other times, `np.int_` worked fine.

### CODE REVIEW

File: tests/test_io.py [hunk 1]
```diff
@@ -66,10 +66,7 @@
 from pandas.io.sas.sas_xport import XportReader
 from pandas.io.stata import StataReader
 
-from . import (
-    lxml_skip,
-    pytables_skip,
-)
+from . import lxml_skip
```
Comment (twoertwein) [inline tests/test_io.py:69]: Could this be `from tests import lxml_skip` (if it is in `__init__.py`)?

**Review by twoertwein — APPROVED** [review]
Looks good to me! Hope you had a somewhat automated approach for all these changes :)

`iloc` vs `__iter__`: I would have been fine keeping the current approach as it is not clear what the "correct" type is.

File: tests/test_io.py (hunk 1 above)
Comment (Dr-Irv) [reply tests/test_io.py:69]: Probably, I just left it as it was.

**Comment by Dr-Irv** [comment]
> Looks good to me! Hope you had a somewhat automated approach for all these changes :)

Unfortunately not!  It was a bit of wack-a-mole.
> 
> `iloc` vs `__iter__`: I would have been fine keeping the current approach as it is not clear what the "correct" type is.

Agreed, but I think we should be doing this level of checking when we get to things like `Int8Dtype()` .

**Comment by twoertwein** [comment]
Thanks @Dr-Irv !

btw, the next mypy release (1.1?) will suggest using pandas-stubs https://github.com/python/mypy/issues/14328#event-8556645254

**Pull request merged**